### PR TITLE
Cleanup configmaps after node/role cleanup

### PR DESCRIPTION
**Pull request by rabi**
This also changes configmap management to use lib-common api.

### CODE REVIEW

File: pkg/deployment/inventory.go [hunk 1]
```diff
@@ -99,40 +97,29 @@ func GenerateRoleInventory(ctx context.Context, helper *helper.Helper,
...
+		"network":   string(instance.Spec.NodeTemplate.NetworkConfig.Template),
+	}
 	configMapName := fmt.Sprintf("dataplanerole-%s", instance.Name)
-	cm := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      configMapName,
-			Namespace: instance.Namespace,
-			Labels:    instance.ObjectMeta.Labels,
+	cms := []utils.Template{
+		// ConfigMap
```
Comment (bshephar) [inline pkg/deployment/inventory.go:111]: Not sure of the purpose of this comment? Should it be more explicit about which configmap is being created? Is the comment needed at all?

File: pkg/deployment/inventory.go [hunk 2]
```diff
@@ -184,40 +171,29 @@ func GenerateNodeInventory(ctx context.Context, helper *helper.Helper,
...
+		"network":   string(networkConfig.Template),
+	}
 	configMapName := fmt.Sprintf("dataplanenode-%s", instance.Name)
-	cm := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      configMapName,
-			Namespace: instance.Namespace,
-			Labels:    instance.ObjectMeta.Labels,
+	cms := []utils.Template{
+		// ConfigMap
```
Comment (bshephar) [inline pkg/deployment/inventory.go:185]: And this one. Do we need a comment?

**Review by bshephar — COMMENTED** [review]
Nice idea to use the common interface. Seems to ensure that we set the `ControllerReference`.

File: pkg/deployment/inventory.go (hunk 1 above)
Comment (rabi) [reply pkg/deployment/inventory.go:111]: Yes there is a puprose. utils.Template can be used for configmaps and secrets and this says it's a configmap[1].

[1] https://github.com/openstack-k8s-operators/infra-operator/blob/main/controllers/memcached/memcached_controller.go#L248

File: pkg/deployment/inventory.go (hunk 2 above)
Comment (rabi) [reply pkg/deployment/inventory.go:185]: as explained above.

**Review by rebtoor — COMMENTED** [review]
+2

**Review by fao89 — APPROVED** [review]

**Pull request merged**